servicetalk-http-router-jersey/src/main/java/io/servicetalk/http/router/jersey/JerseyRouteExecutionStrategy.java

```diff
@@ -74,14 +74,11 @@ public boolean isSendOffloaded() {
 
     @Override
     public HttpExecutionStrategy merge(final HttpExecutionStrategy other) {
-        return delegate.merge(other);
+        HttpExecutionStrategy merged = delegate.merge(other);
+        return merged == delegate ? this : new JerseyRouteExecutionStrategy(merged, executor);
```

Comment:
seems we are getting more specialized in `merge`, does this mean we won't be able to make `merge` a non-instance method?